client/nearby/src/components/WorkspaceContainer.jsx

```diff
@@ -2,30 +2,26 @@
 import React from 'react';
 import Workspace from './workspace/Workspace';
 import Title from './Title';
-import { getWorkspaces } from '../actions';
+import { getNearbyWorkspaces } from '../actions';
 
-const WorkspaceContainer = (props = null) => {
-    const [locs, setLocs] = React.useState(props.nearbyWorkspaces || null);
-    const [allInfo, setAllInfo] = React.useState(props.allWorkspaceInfo || []);
-    const [pic, setPic] = React.useState(props.photos || []);
+const WorkspaceContainer = (props) => {
+    const [locs, setLocs] = React.useState(props.nearbyWorkspaces || []);
 
     React.useEffect(() => {
-        if (!props) {
-            getWorkspaces()
-                .then(({ nearbyWorkspaces, allWorkspaceInfo, photos }) => {
+        if (locs.length === 0) {
+            getNearbyWorkspaces()
+                .then(({ nearbyWorkspaces }) => {
                     setLocs(nearbyWorkspaces);
-                    setAllInfo(allWorkspaceInfo);
-                    setPic(photos);
                 })
-                .catch(() => setLocs(false));
+                .catch(() => setLocs([]));
         }
     }, []);
 
-    return locs ? (
+    return locs.length > 0 ? (
         <React.Fragment>
             <Title />
-            {locs.map((location) => (
-                <Workspace key={location.workspaceId} location={location} allInfo={allInfo} pic={pic} />
+            {locs.map((loc, i) => (
+                <Workspace key={i} location={loc} />
```

Comment:
Using workspaceId is better here. `key` should be a unique id so React can make efficient updates. Index only works if the array isn't modified. If it is, the index of the item changes and React gets confused: https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
Easy fix for this one, good looking out!